hist/hist/src/TProfile.cxx

```diff
@@ -416,6 +416,25 @@ Int_t TProfile::BufferFill(Double_t x, Double_t y, Double_t w)
    return -2;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Run a Chi2Test between two TProfiles.
+/// This calls TH1::Chi2Test() with the option "WW".
+
+Double_t TProfile::Chi2Test(const TH1 *h2, Option_t *option, Double_t *res) const
+{
+   TString opt = option;
+   opt.ToUpper();
+   opt += "WW";
```

Comment:
should this happen only if h2 is a TProfile?
Actually, does it make sense to compute a chi2 between a TProfile and a non-TProfile?

Reply:
ok, maybe you can consider overloading only the case where the other histogram is actually a `TProfile` and not a generic `TH1`.

Reply:
I considered this, but an overload `TProfile::Chi2Test(TProfile*, ...)` could not be called via a `TH1` pointer. So this one
```c++
TH1* profile = new TProfile(...);
profile->Chi2Test( new TProfile(...) );
```
would call `TH1::Chi2Test(TH1*)`. This might be OK if users use their TProfiles always as TProfiles, but maybe polymorphism isn't such a bad thing here, when you e.g. want to collect histograms and profiles in the same container.
For this reason, I tend towards overriding instead of overloading. That would get us back to casting `h2`, and if the cast succeeds, the options get set to `WW`. If it fails, the users can choose any option they desire, like you did with the original version of the function.